File: app/src/main/java/bkm/com/newsapp/ui/list/MainViewModelFactory.java
package bkm.com.newsapp.ui.list;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import bkm.com.newsapp.data.NewsRepository;

public class MainViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private final NewsRepository mRepository;

    public MainViewModelFactory(NewsRepository repository) {
        this.mRepository = repository;
    }

    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new MainActivityViewModel(mRepository);
    }
}

File: app/src/main/java/bkm/com/newsapp/ui/list/MainViewModelFactory.kt
package bkm.com.newsapp.ui.list

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import bkm.com.newsapp.data.NewsRepository

class MainViewModelFactory(val newsRepository: NewsRepository) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MainActivityViewModel(newsRepository) as T
    }
}